src/query/streams.cpp

```diff
@@ -434,4 +456,14 @@ void Streams::Persist(StreamStatus &&status) {
 }
 
 std::string_view Streams::BootstrapServers() const { return bootstrap_servers_; }
+
+std::string Streams::SetStreamOffset(const std::string_view stream_name, int64_t offset) {
+  auto lock_ptr = streams_.Lock();
+  if (auto it = lock_ptr->find(std::string(stream_name)); it != lock_ptr->end()) {
+    auto consumer_lock_ptr = it->second.consumer->Lock();
+    return consumer_lock_ptr->SetConsumerOffsets(offset);
+  }
+  return fmt::format("Stream: {} not found", stream_name);
```

Comment:
```suggestion
  auto it = GetStream(*locked_streams, stream_name);
  auto consumer_lock_ptr = it->second.consumer->Lock();
  return consumer_lock_ptr->SetConsumerOffsets(offset);
```

I don't like `std::string` return value for error handling in general, but here there is an extra argument against it: every other function throws an exception if the stream cannot be found. This function breaks the consistency.